packages/devtools_app/lib/src/screens/logging/logging_screen_v2/logging_table_v2.dart

```diff
@@ -243,75 +252,104 @@ class LoggingSettingsDialogV2 extends StatelessWidget {
       ],
     );
   }
 }
 
-/// Shows and hides the context menu based on user gestures.
-///
-/// By default, shows the menu on right clicks and long presses.
-class _ContextMenuRegion extends StatefulWidget {
-  /// Creates an instance of [_ContextMenuRegion].
-  const _ContextMenuRegion({
-    required this.child,
-    required this.contextMenuBuilder,
+class _RententionLimitSetting extends StatefulWidget {
+  const _RententionLimitSetting({
+    required this.theme,
+    required this.onRetentionLimitChange,
   });
 
-  /// Builds the context menu.
-  final ContextMenuBuilder contextMenuBuilder;
-
-  /// The child widget that will be listened to for gestures.
-  final Widget child;
+  final ThemeData theme;
+  final void Function(int) onRetentionLimitChange;
 
   @override
-  State<_ContextMenuRegion> createState() => _ContextMenuRegionState();
+  State<_RententionLimitSetting> createState() =>
+      _RententionLimitSettingState();
 }
 
-class _ContextMenuRegionState extends State<_ContextMenuRegion> {
-  final _contextMenuController = ContextMenuController();
-
-  void _onSecondaryTapUp(TapUpDetails details) {
-    _show(details.globalPosition);
-  }
-
-  void _onTap() {
-    if (!_contextMenuController.isShown) {
-      return;
-    }
-    _hide();
-  }
-
-  void _show(Offset position) {
-    _contextMenuController.show(
-      context: context,
-      contextMenuBuilder: (BuildContext context) {
-        return widget.contextMenuBuilder(context, position);
-      },
-    );
+class _RententionLimitSettingState extends State<_RententionLimitSetting>
+    with AutoDisposeMixin {
+  void updateRetentionLimit() {
+    newRetentionLimit = preferences.logging.retentionLimit.value;
   }
 
-  void _hide() {
-    _contextMenuController.remove();
+  @override
+  void initState() {
+    super.initState();
+    preferences.logging.retentionLimit.addListener(updateRetentionLimit);
+    updateRetentionLimit();
   }
 
   @override
   void dispose() {
-    _hide();
+    preferences.logging.retentionLimit.removeListener(updateRetentionLimit);
     super.dispose();
   }
 
+  late int newRetentionLimit;
+
   @override
   Widget build(BuildContext context) {
-    return GestureDetector(
-      behavior: HitTestBehavior.opaque,
-      onSecondaryTapUp: _onSecondaryTapUp,
-      onTap: _onTap,
-      child: widget.child,
+    return Row(
+      children: [
+        Expanded(
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              Text(preferences.logging.retentionLimitTitle),
+              Text(
+                'Used to limit the number of log messages retained.',
+                style: widget.theme.subtleTextStyle,
+              ),
+            ],
+          ),
+        ),
+        const SizedBox(width: defaultSpacing),
+        SizedBox(
+          height: defaultTextFieldHeight,
+          width: defaultTextFieldNumberWidth,
+          child: TextField(
```

Comment:
looking at the screenshot, the text is not vertically centered in the text box. Can we fix that as part of this change?

Reply:
<img width="457" alt="Screenshot 2024-06-24 at 1 35 40 PM" src="https://github.com/flutter/devtools/assets/1386322/aca3048c-80c4-485c-809a-f67e1fe51e9a">

To get it closer to vertically centered, I've set it to TextAlignVertical.top
a